docs: landing and architecture page updates (#26)
armory/docs-cdaas: content/en/architecture.md

---
title: Armory CD-as-a-Service Architecture
linkTitle: Architecture
weight: 10
description: >

>   In this high-level overview, learn how Armory Continuous Deployment-as-a-Service works and about its key components.

Reviewer: ```suggestion
 Learn how Armory Continuous Deployment-as-a-Service works and how its key components orchestrate continuous deployment to your Kubernetes clusters.
```

I changed my mind

---

{{< include "cdaas-explained-how.md" >}}

## Key components

### Armory CD-as-a-Service API (Control Plane)
The control plane is the set of services comprising the Armory CD-as-a-Service APIs.

> It's where the business logic for CD-as-a-Service lives, and it utilizes Remote Network Agents to talk to customer's privately networked resources such as Kubernetes APIs, Jenkins, Prometheus, etc., as well as external services, such as New Relic, Datadog, and more.

Reviewer: What do you think of:

The continuous deployment control plane is the set of services comprising the Armory CD-as-a-Service APIs.

Or 
The CD-as-a-Service control plane is the set of services that comprise the APIs.

We should be consistent since "control plane" is used further down. Having worked in AT&T Network - Research, I think "CD-as-a-Service control plane" is clear enough to differentiate it from the K8s control plane or a network control plane. Thoughts?

> It's where the business logic for CD-as-a-Service lives, and it utilizes Remote Network Agents to talk to customer's privately networked resources such as Kubernetes APIs, Jenkins, Prometheus, etc., as well as external services, such as New Relic, Datadog, and more.

Reviewer: Do we need "It's where the business logic for CD-as-a-Service lives" ? Isn't that concept inherent in an API?

I'd combine L14-16.  Something like:

The CD-as-a-Service control plane is the set of services comprising the APIs. This control plane uses a Remote Network Agent to talk to your privately networked resources such as Kubernetes APIs, Jenkins, and Prometheus.  Furthermore, the RNA can talk to external services like New Relic and Datadog.

Couple of points:
1. Address the reader as "you" -> **your** networked resources rather than **customer's** https://docs.armory.io/contribute/create/guides/docs-style-guide/#address-the-reader-as-you
2. Avoid using "etc", "and more" -> list the items (https://developers.google.com/style/word-list#letter-e) (scroll to the "etc" entry)

### Remote Network Agent (RNA)

> The RNA is a logic-less network relay that enables Armory CD-as-a-Service to integrate with privately networked resources such as Jenkins, Prometheus, Kubernetes clusters, etc.

Reviewer: ```suggestion
The RNA is a logicless network relay that enables CD-as-a-Service to integrate with privately networked resources such as Jenkins, Prometheus, and Kubernetes clusters.
```
Generally we don't use "etc" -> list out the options. "such as" indicates a list of examples. Avoid using "etc", "and more" -> list the items (https://developers.google.com/style/word-list#letter-e)

> As an enhancement for Kubernetes, the Control Plane uses the agent's Service Account credentials to automatically register any cluster it's installed in as a deployable target.

Reviewer: ```suggestion
For Kubernetes, the CD-as-a-Service control plane uses an RNA's ServiceAccount credentials to automatically register the cluster the RNA is installed in as a deployment target.
```

Still not happy with this sentence - bit too long

How to refer to Kubernetes objects https://docs.armory.io/contribute/create/guides/docs-style-guide/#kubernetes-objects, which directs you to https://kubernetes.io/docs/contribute/style/style-guide/#use-upper-camel-case-for-api-objects. _When you refer specifically to interacting with an API object, use [UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case), also known as Pascal case. You may see different capitalization, such as "configMap", in the [API Reference](https://kubernetes.io/docs/reference/kubernetes-api/). When writing general documentation, it's better to use upper camel case, calling it "ConfigMap" instead._

> Once you install the RNA in your cluster, you don't need to update it beyond security updates since deployment logic is encapsulated in Armory's centralized control plane.

Reviewer: ```suggestion
Once you install the RNA in your cluster, you don't need to update it beyond security updates since deployment logic is encapsulated in CD-as-a-Service's centralized control plane.
```

> See the [Remote Network Agent](/remote-network-agent/overview) section of the docs for more details.

Reviewer: ```suggestion
See the [Remote Network Agent]({{< ref "remote-network-agent/overview.md" >}}) overview for details.
```

We don't use hard-coded URLs. Use the `ref` shortcode instead. https://docs.armory.io/contribute/create/guides/tips-tricks/#link-to-other-pages-in-the-docs

OR you can use the `linkWithTitle` shortcode https://docs.armory.io/contribute/create/guides/shortcodes/#hyperlink-with-page-title
```
See {{< linkWithTitle "remote-network-agent/overview.md" >}} for details.
```

### Command Line Interface (CLI)

> The CLI is the primary means of interacting with the Armory CD-as-a-Service.

Reviewer: ```suggestion
The CLI is the primary means of interacting with CD-as-a-Service.
```

> Users and Machines can use the CLI directly or use it in CI to automate deployments.

Reviewer: ```suggestion
You can use the CLI directly or with machine-to-machine credentials to automate deployments in your CI system.
```

> We distribute the CLI as native binaries (amd64 and arm64) for Linux, Mac, and Windows as well as docker image.

Reviewer: ```suggestion
Armory distributes the CLI as native binaries (amd64 and arm64) for Linux, Mac, and Windows as well as a Docker image.
```

- Avoid using "we" https://docs.armory.io/contribute/create/guides/docs-style-guide/#avoid-using-we
- "Docker" has a capital D https://docs.armory.io/contribute/create/guides/docs-style-guide/#armory-word-list

> See the [CLI](/cli) section of the docs for more details.

Reviewer: See {{< linkWithTitle "cli.md" >}} for details.

### Cloud Console (UI)

> Cloud Console is the browser-based UI for Armory CD-as-a-Service. It allows users to visually monitor and interact with deployments. Admins can use it to configure integrations with external services like New Relic and Datadog, secrets and more. 

Reviewer: ```suggestion
_Cloud Console_ is the browser-based UI for CD-as-a-Service. You can visually monitor and interact with your deployments on the **Deployments** screen. If you're an Admin, you can use the **Configure** screens to perform tasks such as:

 - Configuring integrations with external services such as Prometheus, New Relic, and Datadog
 - Creating machine-to-machine credentials
 - Creating secrets
 - Configuring RBAC
 - Inviting users
 - Monitoring your Remote Network Agents
```
